### PR TITLE
Git committed yep

### PR DESCRIPTION
**Pull request by GammaSqu**
This is just an attempt of Homework 13 - have not done the challenge part.

The CSS may not look exactly from the original website, but I had to use a video to see how it was approached and used it for this homework.

Please let me know if there are any issues with this, and better ways to improve the JS.

Thank you!

### CODE REVIEW

File: Week-13/homework-13/homework13.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  let questionNumber = currentQuestionIndex + 1;
+
+  questionElement.innerHTML =
+    'Question ' + questionNumber + '. ' + currentQuestion.question;
+
+  //Adding the answers as buttons
+  currentQuestion.answers.forEach((answer) => {
+    let button = document.createElement('button');
+    button.innerHTML = answer.text;
+    button.classList.add('btn');
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:89]: You can chain multiple class in here
```
button.classList.add('btn', 'btn-outline-secondary', 'd-block')
```

File: Week-13/homework-13/homework13.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+];
+
+//For video referening --> myNames = videoNames
+//quizData = questions
+//questionElement = questionElement
+//answerButtons = answerButtons
+//nextButton = nextButton
+
+// Selecting the question, question number, answer buttons, and next button
+let questionElement = document.querySelector(`[data-component="question"]`);
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:56]: If you do not plan to use multiple lines or use template inside like `my-button="${name}"` then you should not use \`, use single-quote instead

File: Week-13/homework-13/homework13.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+//Creating some function that would load corresponding question number and question
+function showQuestion() {
+  resetState();
+
+  let currentQuestion = quizData[currentQuestionIndex];
+  let questionNumber = currentQuestionIndex + 1;
+
+  questionElement.innerHTML =
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:82]: Use string template to make the whole thing easier to read
```
`Question ${questionNumber}. ${currentQuestion.question}`
```

File: Week-13/homework-13/homework13.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    'Question ' + questionNumber + '. ' + currentQuestion.question;
+
+  //Adding the answers as buttons
+  currentQuestion.answers.forEach((answer) => {
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:86]: That's interesting that you are using `.forEach`, where did you learn it from?

File: Week-13/homework-13/homework13.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    let button = document.createElement('button');
+    button.innerHTML = answer.text;
+    button.classList.add('btn');
+    button.classList.add('btn-outline-secondary');
+    button.classList.add('d-block');
+    answerButtons.appendChild(button);
+    //Adding click event when click an answer (also don't understand what "dataset" is)
+    if (answer.correct) {
+      button.dataset.correct = answer.correct;
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:95]: I would recommend not to save if the answer into the state of the button as ppl who knows tech can inspect your DOM to figure out the answer.
If you want to save information to the button to retrieve it later, you can save the current `index` of the button when looping through the list of answers instead.

Also, do not use `dataset`, I would recommend using `setAttribute` like `setAttribute('data-index', index)`

When user clicks the button, you can get this attribute out then cross check with `quizData` and `currentQuesitonIndex` to know if it's the correct answer or not

File: Week-13/homework-13/homework13.js [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    }
+    button.addEventListener('click', selectAnswer);
+  });
+}
+
+//Adding a reset function - when moving to new question it moves previous asnwers
+function resetState() {
+  nextButton.style.display = 'none';
+  while (answerButtons.firstChild) {
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:104]: Is this from chatGPT?

File: Week-13/homework-13/homework13.js [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    if (answer.correct) {
+      button.dataset.correct = answer.correct;
+    }
+    button.addEventListener('click', selectAnswer);
+  });
+}
+
+//Adding a reset function - when moving to new question it moves previous asnwers
+function resetState() {
+  nextButton.style.display = 'none';
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:103]: Why don't you use `classList.add('d-none')` instead?

File: Week-13/homework-13/homework13.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+  let isCorrect = selectBtn.dataset.correct;
+  if (isCorrect === 'true') {
+    selectBtn.classList.add('btn-success');
+    //Increase the score by one when correct
+    score++;
+  } else {
+    selectBtn.classList.add('btn-danger');
+  }
+  //Not sure what it's adding here - need to revisit!!!
+  Array.from(answerButtons.children).forEach((button) => {
```
Comment (jack-vo) [inline Week-13/homework-13/homework13.js:122]: Interesting, another usage of ChatGPT, it's good that you try to find the answer from ChatGPT, however, I would strongly recommend using our generic `for...loop` for Array first

**Review by jack-vo — COMMENTED** [review]
Good work trying to solve this homework!
I left some comments to improve your work, feel free to review them

File: Week-13/homework-13/homework13.js (hunk 7 above)
Comment (GammaSqu) [reply Week-13/homework-13/homework13.js:103]: I didn't think about that - I was just following the tutorial to get the idea of how to make the quiz

File: Week-13/homework-13/homework13.js (hunk 1 above)
Comment (GammaSqu) [reply Week-13/homework-13/homework13.js:89]: I did try to do this on my first attempt, but it didn't work. After working on it now, somehow it did....
Thanks for that

File: Week-13/homework-13/homework13.js (hunk 2 above)
Comment (GammaSqu) [reply Week-13/homework-13/homework13.js:56]: Ok! Will keep that in mind

File: Week-13/homework-13/homework13.js (hunk 3 above)
Comment (GammaSqu) [reply Week-13/homework-13/homework13.js:82]: Thank you very much

File: Week-13/homework-13/homework13.js (hunk 4 above)
Comment (GammaSqu) [reply Week-13/homework-13/homework13.js:86]: I've learnt it from a YouTube video, I'm not sure what it really does, but it kinda almost looks like the for...loop function. I'm just not sure if they're similar

File: Week-13/homework-13/homework13.js (hunk 6 above)
Comment (GammaSqu) [reply Week-13/homework-13/homework13.js:104]: Nope, from a Youtube video

**Pull request merged**